v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/datasource/DataSource.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.source.reader.io.datasource;
+
+import com.google.auto.value.AutoOneOf;
+import com.google.cloud.teleport.v2.source.reader.io.cassandra.iowrapper.CassandraDataSource;
+
+@AutoOneOf(DataSource.Kind.class)
+public abstract class DataSource {
+  public enum Kind {
+    JDBC,
+    CASSANDRA
```

Comment:
this class would need to be edited for every new data source added. This is suboptimal.

Reply:
But as against that, we have the sources and their methods completely independent.
It's hard to have an inheritance between JDBC datasource, Cassandra Datasource and say dynamo datasource.